### DISCUSSION TITLE
Messages/Notifications not appearing with latest neovim master

### DISCUSSION THREAD

**Discussion by moonstruckmoth** [LazyVim/LazyVim#2544] [Q&A] [answered]
With latest master of neovim, I'm not seeing any messages/notifications.

eg 
:echo "hello"

or 

1C-G (to display full filename)

If I use the stable branch of master, everything works fine. 
Is anyone else facing this issue?

**Comment by dpetka2001** [comment]
With latest nightly

![test](https://github.com/LazyVim/LazyVim/assets/12776461/f2808fe3-2693-4efb-bcff-c2c71ac038e9)

**Reply by moonstruckmoth** [reply]
I didn't try with the latest nightly, but if I build with release-0.9, it works fine.

The master nvim version is:
NVIM v0.10.0-dev-2358+gce5a9bfe7
Build type: Release
LuaJIT 2.1.1707061634

I tried the master builds on three machines and all have the same behavior, ie, if I build with latest master, messages don't appear.

I will try git bisect sometime later to figure out the commit causing this behavior

**Comment by moonstruckmoth** [comment]
git bisect tells me the following is the commit from which messages started disappearing. I've not analyzed the changes yet tho'


commit b162adbb7ca4f980e938a4a06a49218a9ed496cf (HEAD)
Author: nikolightsaber <103886134+nikolightsaber@users.noreply.github.com>
Date:   Thu Feb 8 11:59:31 2024 +0100

    feat(api): pass 0 to nvim_get_chan_info for current channel (#27321)

    Getting current channel info was kind of annoying via RPC. Two
    functions had to be called:
    1. `nvim_get_api_info` which returns `[channel_id, meta_data]`.
      - This results in `channel_id = api.nvim_get_api_info()[0]`.
      - Here the meta_data is sent but never used.
    2. Finally call `nvim_get_chan_info(channel_id)`.

    This commit reduces the need for `nvim_get_api_info` as passing 0
    returns current channel info.

 runtime/doc/api.txt              | 3 +++
 runtime/doc/news.txt             | 2 ++
 runtime/lua/vim/_meta/api.lua    | 2 +-
 src/nvim/api/vim.c               | 8 +++++++-
 test/functional/api/vim_spec.lua | 4 +++-
 5 files changed, 16 insertions(+), 3 deletions(-)

**Reply by dpetka2001** [reply]
That commit was 5 days ago, which means that it must be included in the latest nightly that I used from this morning (commit hash `1c7b0b9`). Still I'm not getting the issue you're observing. I'm using `bob` to install the latest nightly. Could you temporarily try installing Neovim via `bob` and test the behavior? Maybe something went amiss during your build process? Not sure.

**Comment by xieyanknockout** [comment]
the same problem, my os is unbuntu22.04, use the ppa:unstable to install ,nvim version is 0.10.0-dev

**Reply by dpetka2001** [reply]
Can you please try with a fresh LazyVim starter template like so
```bash
1. `mkdir ~/.config/lazytest`
2. `git clone https://github.com/LazyVim/starter ~/.config/lazytest`
3. `NVIM_APPNAME=lazytest nvim` and wait for the installation to complete
4. Restart Neovim with the command provided in step 3
```
Do you observe the same behavior?

For cleanup:
```bash
rm -rf ~/.config/lazytest
rm -rf ~/.local/share/lazytest
rm -rf ~/.local/state/lazytest
rm -rf ~/.cache/lazytest
```

**Reply by moonstruckmoth** [reply]
I tried the steps mentioned above and I can see the same behavior, ie messages/notifications are not displayed.
I tested this with echo "hello" and 1C-G.

It doesn't seem like a plugins issue because for stable and commits prior to 162adbb7, things work fine.
We can go back and forth between the builds and the behavior will be seen or not seen depending on whether the problematic commit is included or not.

(BTW, I did not try excluding only that particular commit)

**Reply by dpetka2001** [reply]
Unfortunately,  I'm not able to reproduce, so I can't help any further and am out of ideas. Sorry for that.

Maybe someone if someone else has experienced the same behavior will be able to give you some more helpful advice.

**Reply by dpetka2001** [reply, marked as answer]
I think i may have found the culprit. Can you try the following
```lua
return {
    {
    "ls-devs/nvim-notify",
    branch = "fix/fix_index_value",
  },
}
```
Then in the Lazy UI find the plugin `nvim-notify` and then press `x` to uninstall it and then `i` to re-install it. There has been some recent changes in Neovim upstream that broke the plugin and there's currently a PR waiting to be merged to fix this. But until it gets merged, you can use the branch from that [PR](https://github.com/rcarriga/nvim-notify/pull/253) to temporarily fix your issue.

**Reply by moonstruckmoth** [reply]
One update:

"nvim --clean" (which skips user config and plugins, shada) doesn't have the issue even with latest master

The problem could be interaction between nvim and plugins like Noice/notify

**Comment by jlvcm** [comment]
My notifications  appear as mostly black, and never disapear, and sometimes i get na error
```
NVIM v0.10.0-dev-2361+ga376d979b-Homebrew
Build type: Release
LuaJIT 2.1.1703358377
Run "nvim -V1 -v" for more info
```

<img width="1028" alt="image" src="https://github.com/LazyVim/LazyVim/assets/1488668/2c4f54df-1bfe-49d9-9bb1-8a6f987f4748">

**Reply by moonstruckmoth** [reply]
I too have seen some notifications (like diagnostics being disabled, which is one of my default settings) appearing in black, but output of things like echo, 1C-G etc is not seen.

**Reply by dpetka2001** [reply]
Did you try what I wrote in my other [post](https://github.com/LazyVim/LazyVim/discussions/2544#discussioncomment-8463751)?

**Reply by moonstruckmoth** [reply]
Yes your suggestion worked. Thanks for taking time to figure this out.

The issue is in the nvim-notify plugin as you mentioned. Till it gets fixed, I'll use the config you suggested.

Thanks again!

**Reply by jlvcm** [reply]
thanks @dpetka2001 that fixes it!

**Reply by moonstruckmoth** [reply]
Looks like the PR has been merged, so we don't need to specify the branch anymore. I tested it just now
